src/components/User/UserComponent.jsx

```diff
@@ -1,120 +1,253 @@
-import { ToastMessage } from '../../_utils/ToastMessage';
+import FilterComponent from '../common/FilterComponent';
+import FilterButton from '../common/FilterButton';
+import StatusMessageComponent from '../common/StatusComponent';
+import { isCountCutoffExceeded, fetchData } from '../../_utils/helpers';
 
-const UserComponent = (props) => {
-  const hasNoUsers = isEmpty(props.activePageUsers);
+import '../../_css/UsersComponent.css';
 
-  if (props.isLoading) {
-    return <LoaderComponent />;
+const CUTOFF_LIMIT = 100;
+const checkIfCutoffExceeded = isCountCutoffExceeded(CUTOFF_LIMIT);
+
+export default class UserComponent extends React.Component {
+  state = {
+    limit: 10,
+    users: [],
+    allDataFetched: false,
+    isFiltered: false
+  };
+
+  componentDidMount() {
+    const usersEmpty = isEmpty(this.props.users);
+    if (usersEmpty) {
+      this.props.loadUsers(this.props.activePage, this.state.limit);
+    }
+    this.props.loadAllFilterValues();
   }
 
-  if (props.hasError && props.errorMessage) {
-    setTimeout(() => {
-      ToastMessage.error({ message: props.errorMessage });
-    }, 500);
-    return <SemanticToastContainer />;
+  handleRowChange = (e, data) => {
+    this.setState({
+      limit: data.value,
+      users: []
+    });
+    this.props.resetUsers();
+    this.retrieveUsers(this.props.activePage, data.value);
+  };
+
+  handlePaginationChange = (e, { activePage }) => {
+    this.props.setActivePage(activePage);
+    const currentPageList = this.props.users[`page_${activePage}`];
+    if (isEmpty(currentPageList)) {
+      this.retrieveUsers(activePage, this.state.limit);
+    }
+  };
+
+  retrieveUsers = (activePage, limit) => {
+    if (checkIfCutoffExceeded(activePage, limit)) {
+      const url = `users?page=${activePage}&page_size=${limit}`;
+      this.props.loading(true);
+      return fetchData(url).then((response) => {
+        this.props.loading(false);
+        this.setState({ users: response.data.results });
+      }).catch(() => {
+        this.props.loading(false);
+        this.setState({ allDataFetched: true });
+      });
+    }
+    return this.props.loadUsers(activePage, limit);
+  };
+
+  handlePageTotal = () => Math.ceil(this.props.usersCount / this.state.limit);
+
+  emptyUsersList = () => (isEmpty(this.props.users));
+
+  filterApplied = () => {
+    this.setState({ isFiltered: !this.state.isFiltered });
   }
 
-  if (hasNoUsers) {
-    return (
-      <NavBarComponent>
-        <ItemsNotFoundComponent allDataFetched={props.allDataFetched} message="Please try again later, to see if we'll have users to show you." />
-      </NavBarComponent>
+  render() {
+    const currentUsers = `page_${this.props.activePage}`;
+    const activePageUsers = this.props.users[currentUsers] || this.state.users;
+    const hasNoUsers = isEmpty(activePageUsers);
+    const showStatus = this.props.hasError && this.props.errorMessage;
+
+    const filterButton = () => (
```

Comment:
Please move this to it's own component. I have created a gist for you illustrating how to do it [here](https://gist.github.com/sachungo/fec45438336a94dcf5884d81c389c024)